app/jobs/scheduled/reindex_search.rb

```diff
@@ -60,8 +61,15 @@ def rebuild_problem_tags(limit = 10000)
 
     private
 
+    def clean_post_search_data
```

Comment:
Putting this in a scheduled job for now instead of a migration because I'm predicting that we'll need to do a bunch of clean up while I look into search.